Hospital.Infrastructure.InMemory/AppointmentInMemoryRepository.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Hospital.Domain.Entities;
+using Hospital.Domain.Interfaces;
+using Hospital.Tests;
+using System.Data;
+
+namespace Hospital.Infrastructure.InMemory;
+
+/// <summary>
+/// In-memory repository for managing <see cref="Appointment"/> entities.
+/// Uses <see cref="TestData"/> as a simple data source for testing and development.
+/// </summary>
+public class AppointmentInMemoryRepository(TestData _seed) : IAppointmentRepository
+{
+    /// <summary>
+    /// Retrieves all appointments from the in-memory collection, ordered by their ID.
+    /// </summary>
+    /// <returns>A collection of all <see cref="Appointment"/> objects.</returns>
+    public IEnumerable<Appointment> GetAll() => _seed.Appointments.OrderBy(x => x.Id);
+
+    /// <summary>
+    /// Retrieves a single appointment by its unique identifier.
+    /// </summary>
+    /// <param name="id">The appointment ID.</param>
+    /// <returns>
+    /// The matching <see cref="Appointment"/> if found; otherwise, <c>null</c>.
+    /// </returns>
+    public Appointment? Get(int id) => _seed.Appointments.FirstOrDefault(x => x.Id == id);
+
+    /// <summary>
+    /// Adds a new appointment to the in-memory collection.
+    /// The appointment ID is automatically generated.
+    /// </summary>
+    /// <param name="a">The <see cref="Appointment"/> instance to add.</param>
+    /// <returns>The added <see cref="Appointment"/> with its assigned ID.</returns>
+    public Appointment Add(Appointment a)
+    {
+        a.Id = (_seed.Patients.Any() ? _seed.Patients.Max(x => x.Id) : 0) + 1;
```

Comment:
Используйте Count вместо Any. Тут и по коду где это необходимо.
```suggestion
        a.Id = (_seed.Patients.Count != 0 ? _seed.Patients.Max(x => x.Id) : 0) + 1;
```